### DISCUSSION TITLE
Running rqlite on Android

### DISCUSSION THREAD

**Discussion by youk** [rqlite/rqlite#1987] [General]
Are there any inherent drawbacks to running rqlite on Android? I compiled a variant of Linux binary which I was able to run in Termux (using [PRoot](https://wiki.termux.com/wiki/PRoot)). Replication seems to work fine, but I didn't test it extensively.

**Comment by otoolep** [comment]
No issues that I know of. Let me know how it goes.

**Reply by youk** [reply]
In general, would you be interested in releasing a binary for Android? Or there's no target audience for it? 

The build procedure looks straightforward. I only had to add a few compiler flags as compares to `aarch64-linux-gnu-gcc`.

**Reply by otoolep** [reply]
Sure, what are the flags?

**Reply by youk** [reply]
What I quickly came up with:

`-buildmode=pie`
`-ldflags="-linkmode=external -extldflags=--static-pie"`

Please note that I am a stranger to this.

**Reply by otoolep** [reply]
OK,I'll need to do some research in that sense, before I create a build with a set of flags.

**Reply by youk** [reply]
Termux project maintains its [own repository](https://github.com/termux/termux-packages/blob/master/packages/) of packages. It might be useful as a reference. [Here's](https://github.com/termux/termux-packages/blob/master/packages/syncthing/build.sh) for example the build script for Syncthing.
